cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -694,5 +694,55 @@ TEST(TestParquetStatistics, NullMax) {
   EXPECT_EQ(stat_expression->ToString(), "(x >= 1)");
 }
 
+// Tests round-trip projection with nested/indexed FieldRefs
+// https://github.com/apache/arrow/issues/35579
+TEST(TestRoundTrip, ProjectedFieldRefs) {
```

Comment:
Yeah, I only did it this way to more or less match the example in the original issue. Using the existing fixtures would certainly be more convenient. AFAIK the only difference is that it would test the intermediate batch rather than doing the full round trip on the projection - but that's consistent with the other tests.

Reply:
Ok.  I don't feel too strongly here.  So feel free to let me know which you prefer.